auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordActivity.java

```diff
@@ -54,51 +56,57 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fui_forgot_password_layout);
 
-        mEmailFieldValidator =
-                new EmailFieldValidator((TextInputLayout) findViewById(R.id.email_layout));
+        mHandler = ViewModelProviders.of(this).get(RecoverPasswordHandler.class);
+        mHandler.init(getFlowHolder());
+        mHandler.getPasswordResetListener().observe(this, new Observer<Task<String>>() {
+            @Override
+            public void onChanged(Task<String> task) {
+                if (task.isSuccessful()) {
+                    mEmailInputLayout.setError(null);
+                    RecoveryEmailSentDialog.show(task.getResult(), getSupportFragmentManager());
+                } else if (task.getException() instanceof FirebaseAuthInvalidUserException
+                        || task.getException() instanceof FirebaseAuthInvalidCredentialsException) {
+                    // No FirebaseUser exists with this email address, show error.
+                    mEmailInputLayout.setError(getString(R.string.fui_error_email_does_not_exist));
+                } else {
+                    mEmailInputLayout.setError(task.getException().getLocalizedMessage());
+                }
+            }
+        });
+        getFlowHolder().getProgressListener().observe(this, new Observer<Boolean>() {
```

Comment:
Again I have a preference for using a more abstracted type than `Boolean` maybe an object like `ProgressState` that has a boolean for if something is in progress?  Makes the purpose clearer.